### PR TITLE
Fix git integration tests for Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?

Fixes integration tests for the git module on Windows

https://github.com/saltstack/zh/issues/853

### CODE REVIEW

**Comment by rallytime** [comment]
@twangboy Looks like this change is causing the following new and related test failure:

- https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/3322/testReport/junit/integration.modules.git/GitModuleTest/test_init/

File: tests/integration/modules/git.py [hunk 1]
```diff
@@ -581,10 +579,13 @@ def test_init(self):
...
-        new_repo = tempfile.mkdtemp(dir=integration.TMP)
+        new_repo = os.path.normcase(tempfile.mkdtemp(dir=integration.TMP))
+        if salt.utils.is_windows():
+            new_repo = new_repo.replace('\\', '/')
+
         self.assertEqual(
-            self.run_function('git.init', [new_repo]),
-            'Initialized empty Git repository in {0}/.git/'.format(new_repo)
+            self.run_function('git.init', [new_repo]).lower(),
+            'initialized empty git repository in {0}/.git/'.format(new_repo)
```
Comment (cachedout) [inline tests/integration/modules/git.py]: This is going to produce inconsistent results. You're going to get mixed case for new_repo on some platforms and then compare it to something that's forced to lower-case.

Comment (twangboy) [inline tests/integration/modules/git.py]: the `os.path.normcase` on line 582 returns an all lowercase path

**Comment by twangboy** [comment]
Go Go Jenkins!

**Pull request merged**